homeassistant/components/comelit/humidifier.py

```diff
@@ -122,61 +122,32 @@ def __init__(
         self._active_action = active_action
         self._set_command = set_command
 
-    @property
-    def _humidifier(self) -> list[Any]:
-        """Return humidifier device data."""
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        device = self.coordinator.data[CLIMATE][self._device.index]
+        if not isinstance(device.val, list):
```

Comment:
Side note: That we need this check is a code smell that something in the library typing or design can be improved.

Reply:
Could be.
The value form the API can vary form `str`, `int`, `float` or `list`.
But for the climate and humidifier is always a `list`.

Open to suggestions.

Reply:
Maybe it's possible to have different type annotations for the different device types and use these appropriately depending on what top level string is used in the dict? Eg here we use `CLIMATE` as the top level string, and then I assume we know that the devices will be of climate/humidifier device type. The library could type this in more detail, either with a TypedDict or with a dataclass.

Reply:
What if we proceed with this PR and then I work on the suggested improvement on library side ?

Reply:
Yes, my side notes are never required to be solved in the current PR. :+1: